core/templates/pages/exploration-editor-page/services/router.service.spec.ts

```diff
@@ -420,7 +409,19 @@ describe('Router Service', function() {
   it('should save pending changes', function() {
     var broadcastSpy = spyOn($rootScope, '$broadcast').and.callThrough();
     RouterService.savePendingChanges();
-    $timeout.flush();
     expect(broadcastSpy).toHaveBeenCalledWith('externalSave');
   });
+
+  it('should save pending changes when function throws an error', function() {
+    spyOn($rootScope, '$broadcast').and.throwError(
+      'Cannot read property $$nextSibling of null');
+    var applySpy = spyOn($rootScope, '$apply').and.callThrough();
+    // $destroy is being mocked here because it is being called
+    // in the test and it calls $broadcast which throws an error on
```

Comment:
I think this needs a bit more explanation. Where is it being called in the test? The comment should point to that.

Reply:
Done. Is the comment clear enough?